qiskit_nature/problems/second_quantization/molecular/molecular_problem.py

```diff
@@ -73,8 +73,18 @@ def second_q_ops(self) -> List[SecondQuantizedOp]:
         return second_quantized_ops_list
 
     def _transform_q_molecule(self, q_molecule) -> QMolecule:
+        ParticleHoleTransformer = None
+
         for transformer in self.transformers:
-            q_molecule = transformer.transform(q_molecule)
+            # ensure Particle Hole Transformer is executed last
+            if transformer.__class__.__name__ == 'ParticleHoleTransformer':
+                ParticleHoleTransformer = transformer
+            else:
+                q_molecule = transformer.transform(q_molecule)
+
+        if ParticleHoleTransformer is not None:
+            q_molecule = ParticleHoleTransformer.transform(q_molecule)
```

Comment:
I thought the intent was to have some sort of priority information in the transformer itself that they could be ordered here according to that priority to execute the transformation. And then whatever was put in as the priority for ParticleHole it could be done such that it would be last. This does work for now, but at the very minimum a TODO could be put in the code to more have ordering info supplied by transformers, whatever that might be.

Also it would be better to just so instanceof rather than - variables there also should not be mixed case - it makes it look like a class name and is kinda confusing - but then I imagine lint will complain too...

Reply:
I think this is fixed now.